### ISSUE TITLE
Getting user info failure caused runtime crash(nil pointer) issue

### ISSUE THREAD

**Issue by steven-zou** [vmware/harbor#4116]
config auth mode to 'uaa'
the hostname of uaa can not be accessed in the container, then failed to get user info and caused runtime nil pointer issue.

**Comment by steven-zou** [comment]
logs:
=======
```
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018-01-24T03:04:35Z [DEBUG] [security.go:286]: user information is nil
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018-01-24T03:04:35Z [DEBUG] [security.go:300]: using local database project manager
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018-01-24T03:04:35Z [DEBUG] [security.go:302]: creating local database security context...
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018-01-24T03:04:35Z [DEBUG] [user.go:304]: Check if user Q0882S@test.org is super user
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018-01-24T03:04:35Z [DEBUG] [authenticator.go:96]: Current AUTH_MODE is uaa_auth
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:854][C] the request url is  /login #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:855][C] Handler crashed with error runtime error: invalid memory address or nil pointer dereference #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/asm_amd64.s:509 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/panic.go:491 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/panic.go:63 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/signal_unix.go:367 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /go/src/github.com/vmware/harbor/src/ui/auth/uaa/uaa.go:84 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /go/src/github.com/vmware/harbor/src/ui/auth/authenticator.go:113 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /go/src/github.com/vmware/harbor/src/ui/controllers/base.go:60 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/asm_amd64.s:509 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/reflect/value.go:434 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/reflect/value.go:302 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /go/src/github.com/vmware/harbor/src/vendor/github.com/astaxie/beego/router.go:778 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/net/http/server.go:2619 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/net/http/server.go:1801 #033[0m
Jan 24 03:04:35 172.18.0.1 ui[14389]: 2018/01/24 03:04:35 #033[1;35m[router.go:861][C] /usr/local/go/src/runtime/asm_amd64.s:2337 #033[0m
Jan 24 03:05:01 172.18.0.1 ui[14389]: 2018-01-24T03:05:01Z [DEBUG] [security.go:286]: user information is nil
Jan 24 03:05:01 172.18.0.1 ui[14389]: 2018-01-24T03:05:01Z [DEBUG] [security.go:300]: using local database project manager
```

**Comment by reasonerjt** [comment]
Fixed in #4118
